commons-impl/src/main/java/com/connexta/transformation/commons/inmemory/status/InMemoryMetadataTransformation.java

```diff
@@ -0,0 +1,178 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.transformation.commons.inmemory.status;
+
+import com.connexta.transformation.commons.api.ErrorCode;
+import com.connexta.transformation.commons.api.RequestInfo;
+import com.connexta.transformation.commons.api.exceptions.TransformationException;
+import com.connexta.transformation.commons.api.status.MetadataTransformation;
+import com.google.common.io.ByteSource;
+import com.google.common.io.ByteStreams;
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.OptionalLong;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class InMemoryMetadataTransformation implements MetadataTransformation {
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(InMemoryMetadataTransformation.class);
+  private final Object stateLock = new Object();
+  private final String metadataType;
+  private final String transformId;
+  private final RequestInfo requestInfo;
+  private final Instant startTime;
+  private Instant completionTime;
+  private byte[] content;
+  private String contentType;
+  private State state;
+  private ErrorCode failureReason;
+  private String failureMessage;
+
+  public InMemoryMetadataTransformation(
+      String metadataType, String transformId, RequestInfo requestInfo) {
+    this.startTime = Instant.now();
+    this.state = State.IN_PROGRESS;
+    this.metadataType = metadataType;
+    this.transformId = transformId;
+    this.requestInfo = requestInfo;
+  }
+
+  @Override
+  public String getMetadataType() {
+    return metadataType;
+  }
+
+  @Override
+  public Optional<InputStream> getContent() {
+    if (content == null) {
+      return Optional.empty();
+    } else {
+      try {
+        return Optional.of(ByteSource.wrap(content).openStream());
+      } catch (IOException e) {
+        LOGGER.error(
```

Comment:
❗️ Good indication that the interface should declare it throws `IOException` and let that error bubble out

Reply:
Looks like Guava was throwing the `IOException`. The plain-old Java way avoids the catch block entirely! 🎉